src/app.c

```diff
@@ -213,18 +218,28 @@ bool MTY_WindowSetUITexture(MTY_App *app, MTY_Window window, uint32_t id, const
 		return false;
 
 	MTY_Device *device = gfx_ctx_get_device(cmn);
+	if (!device)
+		return false;
 
 	bool r = gfx_begin_ui(cmn, device);
 
 	if (r) {
-		void *texture = MTY_HashPopInt(cmn->ui_textures, id);
-		GFX_UI_API[cmn->api].destroy_texture(cmn->gfx_ui, &texture, device);
+		void *texture = MTY_HashGetInt(cmn->ui_textures, id);
+
+		if (rgba) {
+			void *texture_new = GFX_UI_API[cmn->api].create_texture(cmn->gfx_ui, device, rgba, width, height);
 
-		if (rgba)
-			texture = GFX_UI_API[cmn->api].create_texture(cmn->gfx_ui, device, rgba, width, height);
+			if (texture_new) {
+				GFX_UI_API[cmn->api].destroy_texture(cmn->gfx_ui, &texture, device);
 
-		if (texture)
-			MTY_HashSetInt(cmn->ui_textures, id, texture);
+				texture = texture_new;
+				MTY_HashSetInt(cmn->ui_textures, id, texture_new);
+			}
+
+		} else {
+			GFX_UI_API[cmn->api].destroy_texture(cmn->gfx_ui, &texture, device);
+			MTY_HashPopInt(cmn->ui_textures, id);
+		}
 
 		r = texture != NULL;
 	}
```

Comment:
The old logic is to first destroy the existing texture for `id` and get rid of it from the textures hash, and then carry on with creating the new texture with the given pixel data `rgba` if it was provide, finally updating the hash node for this `id` with the new texture handle. 

The problem with this is that when `create_texture` fails and/or returns a null texture handle, the existing perfectly-fine texture is removed. This failure can occur when the GPU resets, and because the entry is removed from the hash indiscriminately, `MTY_WindowHasUITexture` thereafter always returns `false` for the given `id`, which is incorrect behavior. 

My fix is to replace the old/existing texture with the new one IFF the new texture was successfully created. After this change, even when the GPU resets, textures are rendered properly because the hash still maintains a reference to the original valid handle.